app.js

```diff
@@ -55,61 +55,122 @@ function exitWithStatus(exitStatus) {
+					logger.error(err);
+				} else {
+					logger.info("Message published to " + topicName + " SNS topic successfully");
+					logger.debug(data);
+				}
+			});
+		}
+	} catch (e){
+		logger.error('Error processing listen notification from database\n'+e);
+	}
+};
+
 pg.connect(conString, function(err, client, done) {
   logger.info("Database connection successful");
   if (err){
     logger.error("database err: " + err);
     done();
   }
   // Return the listen notification
-  client.on('notification', function(msg) {
-    try {
-      logger.info('Msg: ' + msg);
-      logger.info('Payload: ' + msg.payload);
-      var notification = JSON.parse(msg.payload);
-      logger.info('Parse successful');
-
-      var topicName = "";
-      if (notification.cards.network === 'facebook'){
-        logger.info('Received card submission via Facebook');
-        topicName = "Facebook";
-      } else if (notification.cards.network === 'telegram') {
-        logger.info('Received card submission via Telegram');
-        topicName = "Telegram";
-      } else if (notification.cards.network === 'twitter') {
-        logger.info('Received card submission via Twitter');
-        topicName = "Twitter";
-      }
-
-      //Construct JSON with relevant details for a confirmation response to be published to SNS topic
-      var jsonMessage = {
-        "language" : notification.cards.language,
-        "username" : notification.cards.username,
-        "implementation_area": notification.cards.report_impl_area,
-        "report_id": notification.cards.report_id
-      };
-      if(topicName !== "")
-      {
-        //Construct message payload
-        var params = {
-          Message: JSON.stringify(jsonMessage),
-          TopicArn: "arn:aws:sns:" + process.env.AWS_REGION + ":" + process.env.ACCOUNTID + ":" + topicName
-        };
-        logger.info("Publishing to " + topicName + " SNS topic");
-        sns.publish(params, function(err, data) {
-          if (err) {
-            logger.error("Error on publishing message to topic" + topicName);
-            logger.error(err);
-          } else {
-            logger.info("Message published to " + topicName + " SNS topic successfully");
-            logger.debug(data);
-          }
-        });
-      }
-    } catch (e){
-      logger.error('Error processing listen notification from database\n'+e);
-    }
-  });
+
+  client.on('notification', handle(msg));
+
   // Initiate the listen query
   client.query("LISTEN watchers");
 });
+
+// Comment to add to commit so that github pull request works.
+// Pull request not working as pgreconnect branch contains
+// same commits even though latest one reverted.
+
+// Handle postgres idle connection error (generated by RDS failover among other possible causes)
+pg.on('error', function(err) {
+	logger.error('Postgres connection error: ' + err);
+
+	logger.info('Enabling caching mode and attempting to reconnect at intervals');
+	reports.enableCacheMode();
```

Comment:
reports object not available in this context, but we don't need caching because without database connection there is no incoming data.